### PR TITLE
fix MPP-2308: Reply to last sender

### PR DESCRIPTION
**Pull request by groovecoder**
This PR fixes MPP-2308.

How to test:
1. `PHONES_ENABLED=0 pytest` should run with no errors nor failures
2. `PHONES_ENABLED=1 pytest` should run with no errors nor failures
3. Set up [end-to-end local phone dev](https://github.com/mozilla/fx-private-relay/blob/main/docs/end-to-end-local-phone-dev.md)
4. Send your local relay number a text message
   * [x] You should receive the text message to your real phone
6. Reply to the text message sent by relay
   * [x] The original sender should receive the reply from the relay number
7. As the original sender, reply to the reply
   * [x] You should receive the reply to the reply to your real phone
8. Repeat as desired

- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Review by jwhitlock — COMMENTED** [review]
When I checked [PR 2330](https://github.com/mozilla/fx-private-relay/pull/2330), I would send a message to my Relay number from my real number (555-867-5309 in this example), and would get a response:

```
[Relay 📲 +15558675309]
Message to check PR 2330
```

With this PR, I get:

```
You can only reply if you allow
Firefox Relay to keep a log of
your callers and text senders
```

There is then a link to relay.firefox.com

* Do I need a second phone number to test relay now?
* Should the link be to my server instead of relay.firefox.com?

**Comment by groovecoder** [comment]
> When I checked [PR 2330](https://github.com/mozilla/fx-private-relay/pull/2330), I would send a message to my Relay number from my real number (555-867-5309 in this example), and would get a response:
> 
> ```
> [Relay 📲 +15558675309]
> Message to check PR 2330
> ```
> 
> With this PR, I get:
> 
> ```
> You can only reply if you allow
> Firefox Relay to keep a log of
> your callers and text senders
> ```
> 
> There is then a link to relay.firefox.com
> 
>     * Do I need a second phone number to test relay now?

Yes. Any message coming **FROM** a Relay customer's real number is now treated as a reply message. So you'll need to send message to your Relay Number from some other number. I use a Google Voice number for it. 

> 
>     * Should the link be to my server instead of relay.firefox.com?

Updated.

**Review by jwhitlock — COMMENTED** [review]
Thanks! I used a second phone number, and got the forwarded message. However, when I replied to my relay mask number, the message "You can only reply if you allow Firefox Relay to keep a log of your callers and text senders." was not delivered. 

Twillio reports the message (with my site URL) was delivered, but dropped by the carrier with [Error 30007](https://www.twilio.com/docs/errors/30007)
 Your message content was flagged as going against carrier guidelines.
 
I use [AT&T ActiveArmor](https://www.att.com/security/). It is possible that it allowed the message with `relay.monitor.com` but blocked it with `my-app.ngrok.io` I have "Fraud Risk" set to "Block". I changed it to "Flag" to try again. I also find that it tends to break my internet connection, so I restarted my phone, which seems to fix it. Great app! However, changing to "Flag" in ActiveArmor did not deliver the message.

When I dropped the URL entirely, the message was delivered.

I'm curious if this is a jwhitlock issue, an AT&T issue, or AT&T + ActiveArmor issue. I'd like to see if others are blocked with an `ngrok.io` URL, and if it works in dev or stage.

Maybe include the URL if in production, but just say "Visit the Relay dashboard" otherwise?

File: api/tests/phones_views_tests.py
Comment (jwhitlock) [inline api/tests/phones_views_tests.py:55]: Nit: Add a default `validate` returning `True` in the fixture?

```python
mocked_twilio_validator.validate = Mock(return_value=True)
```

You could then delete that line from tests using `mocked_twilio_validator`. For the tests returning `False`, you'd need:

```python
mocked_twilio_validator.validate.return_value = False
```

File: api/tests/phones_views_tests.py
Comment (jwhitlock) [inline api/tests/phones_views_tests.py:682]: This test doesn't use `mocked_twilio_validator`, so I think this is unneeded

File: api/tests/phones_views_tests.py
Comment (jwhitlock) [inline api/tests/phones_views_tests.py:707]: Same, ` mocked_twilio_validator` is not passed as a fixture

File: api/tests/phones_views_tests.py
Comment (jwhitlock) [inline api/tests/phones_views_tests.py:728]: and again.

Unless the test is supposed to use the `mocked_twilio_validator` fixture, and there is some weirdness going on...

File: api/views/phones.py
Comment (jwhitlock) [inline api/views/phones.py:521]: Good catch!

File: emails/models.py
Comment (jwhitlock) [inline emails/models.py:420]: Nit: change to a one-liner?:

```suggestion
    return relay_number.user.profile_set.get().store_phone_log
```

File: phones/models.py
Comment (jwhitlock) [inline phones/models.py:23]: The `logger` appears unused - remove it?

**Review by jwhitlock — APPROVED** [review]
Thanks @groovecoder! The steps worked for me, once I removed the URL from the "You can't reply" message. I'll leave it up to you if you want to do something about that now, or see if it replicates on the dev server.

There's a few nits and code cleanup suggestions, but they are in tests, not the application code. Feel free to re-ping me if you want a second review after cleanup.

File: phones/models.py
Comment (jwhitlock) [inline phones/models.py:88]: Maybe use `latest`?

```suggestion
        .latest("last_inbound_date")
```

Comment (groovecoder) [reply phones/models.py:88]: Aha! That's why we should have non-null fields!! :trollface:

File: api/tests/phones_views_tests.py
Comment (groovecoder) [reply api/tests/phones_views_tests.py:59]: maybe easier after `RealPhone` is always included? got this when I started down this route:
image

Comment (groovecoder) [reply api/tests/phones_views_tests.py:59]: because we have `if settings.PHONES_ENABLED: from phones.models import RealPhone`

**Comment by groovecoder** [comment]
@jwhitlock I think this is ready for final review!

**Review by jwhitlock — APPROVED** [review]
Thanks for the changes @groovecoder! Please run `black` and then merge!

**Pull request merged**
